### PR TITLE
ns-api: openvpnrw, add download all user configurations

### PR DESCRIPTION
**Pull request by stephdl**
Make a ns-api wrapper to create a bundle of all certficates  in an archive 

Refs: https://github.com/orgs/NethServer/projects/10/views/2?pane=issue&itemId=80692610

```
root@NethSec:~# echo '{"instance": "ns_roadwarrior1"}'| /usr/libexec/rpcd/ns.ovpnrw call download_all_user_configurations
{"archive_path": "/var/run/ns-api-server/downloads/ns_roadwarrior1_user_configurations.tar.gz"}
```

This will create the archive inside: `/var/run/ns-api-server/download`.

if we untar the archive

```
root@NethSec:~# tar -xvzf /var/run/ns-api-server/downloads/ns_roadwarrior1_user_configurations.tar.gz
helene.ovpn
stephane.ovpn
```

### CODE REVIEW

File: packages/ns-api/files/ns.ovpnrw [hunk 1]
```diff
@@ -776,6 +777,32 @@ def download_user_certificate(ovpninstance, username):
...
+                
+                if os.path.exists(ca_cert_path):
+                    tar.add(ca_cert_path, arcname=f"{username}_ca.crt")
+                if os.path.exists(user_cert_path):
+                    tar.add(user_cert_path, arcname=f"{username}.crt")
+                if os.path.exists(user_key_path):
+                    tar.add(user_key_path, arcname=f"{username}.key")
+    # Set the file permissions to 400 (read-only for the owner)
+    os.chmod(tar_file_path, 0o0400)
+    return (f"Certificates for OpenVPN instance {ovpninstance} have been created, you can retrieve them {tar_file_path}.")
```
Comment (gsanchietti) [inline packages/ns-api/files/ns.ovpnrw:804]: An API must always return a JSON, just return something like:
```
{"archive_path": "..."}
```

**Review by gsanchietti — COMMENTED** [review]
The code seems good to me, but please add the API documentation to the README

File: packages/ns-api/README.md [hunk 2]
```diff
@@ -1293,6 +1293,22 @@ Response example for a server in bridged mode:
 
 After `add-instance`, the `ns_description` field is empty. If the field is empty, the instance has never been edited from UI.
 
+### download all user certificates
+
+prepare a tar.gz archive with all user certificates
```
Comment (gsanchietti) [inline packages/ns-api/README.md:1298]:
```suggestion
Prepare a tar.gz archive with all user certificates
```

File: packages/ns-api/README.md [hunk 3]
```diff
@@ -1293,6 +1293,22 @@ Response example for a server in bridged mode:
...
+```
+
+Response example:
+```json
+{
+  "archive_path": "/var/run/ns-api-server/download/ns_roadwarrior1_certificates.tar.gz"
+}
+```
+
+The archive can be exported by scp for example
```
Comment (gsanchietti) [inline packages/ns-api/README.md:1310]:
```suggestion
The archive can be download using `scp` or the files APIs of nethsecurtity-api-server

**Review by gsanchietti — APPROVED** [review]

**Review by gsanchietti — COMMENTED** [review]
Please fix the documentation.

**Comment by cotosso** [comment]
Just tested , it works exactly as expected, the command creates the tar.gz with all the accounts configurations inside.

**Comment by gsanchietti** [comment]
Wait for the fix until all other issues have been released.

**Pull request merged**